### PR TITLE
feat(carousel): hide prev/next if not loop and indexes at bounds

### PR DESCRIPTION
**Pull request by plondon**

### CODE REVIEW

File: template/carousel/carousel.html [hunk 1]
```diff
@@ -1,10 +1,10 @@
 <div ng-mouseenter="pause()" ng-mouseleave="play()" class="carousel" ng-swipe-right="prev()" ng-swipe-left="next()">
   <div class="carousel-inner" ng-transclude></div>
-  <a role="button" href class="left carousel-control" ng-click="prev()" ng-show="slides.length > 1">
+  <a role="button" href class="left carousel-control" ng-click="prev()" ng-show="slides.length > 1" ng-hide="active === 0 && noWrap()">
```
Comment (icfantv) [inline template/carousel/carousel.html]: these changes add two extra watches per slide - can you please investigate combining the ng-show/ng-hide?

**Comment by RobJacobs** [comment]
I'd prefer ng-if over ng-hide/show.  We also need to consider, does it make for a jarring experience with elements being removed rather than just disabling them?

**Comment by plondon** [comment]
Users can always override the display: none !important with a more specific display: block !important; opacity: 0.5 if they want.

**Comment by plondon** [comment]
Also a fan of disable, that way users can choose the more jarring experience if they want with CSS. Since `disabled` is not a valid attribute for an anchor tag is it alright to swap `<a role="button">` for the regular `<button>`?

**Comment by icfantv** [comment]
@plondon are there accessibility issues with doing that or is that the preferred mechanism?  IIRC, using the `btn-link` class had accessibility ramifications.

**Comment by plondon** [comment]
@icfantv `<button>` has the aria `role="button"` by default, so it should act as `<a role="button">` from an accessibility perspective.

**Comment by realalexhomer** [comment]
@plondon @icfantv Yes that is correct, screen readers prefer <button> over the a tag so it is better to use it when possible

**Comment by plondon** [comment]
Since previous and next buttons are "disabled" by the noWrap logic I think we can leave them as anchor tag's.

Instead of `button disabled="true"` as we were discussing above I used `isPrevDisabled()` and `isNextDisabled()` to toggled a `disabled` class, leaving it up to the user to hide, show, fade, or whatever the button's him/herself.

I'll squash these commits if you all agree with this approach.

**Comment by wesleycho** [comment]
This looks solid to me - anyone else have thoughts?

**Comment by RobJacobs** [comment]
LGTM

**Pull request closed without merging**